### PR TITLE
feat: Allow choosing a Node version in the run job

### PR DESCRIPTION
**Pull request by deiga**
Addresses #475

### CODE REVIEW

**Comment by deiga** [comment]
@jennifer-shehane Could you or someone else re-trigger the pipeline, since my user doesn't have permissions to the contexts in the Pipeline :)

**Comment by jennifer-shehane** [comment]
@deiga Someone from our team would have to followup on publishing the changes.

**Review by mschile — APPROVED** [review]

File: .circleci/test-deploy.yml [hunk 1]
```diff
@@ -65,6 +65,13 @@ workflows:
           working-directory: examples/npm-install
           cypress-cache-key: cypress-cache-{{ arch }}-{{ checksum "examples/npm-install/package.json" }}
           post-install: "npm run say-hello && npm run say-goodbye"
+      - cypress/run:
+          filters: *filters
+          name: Custom Node Version Example
+          working-directory: examples/npm-install
+          cypress-cache-key: cypress-cache-{{ arch }}-{{ checksum "examples/npm-install/package.json" }}
+          node-version: "20.15.1"
+          post-install: "npm run say-hello && npm run say-goodbye"
```
Comment (ramykl) [inline .circleci/test-deploy.yml:74]: Should we add something here that actually makes sure it gets the correct node version?
Something like:
```
if ! node --version | grep -q "20.15.1"; then
    echo "Node version 20.15.1 not found"
    exit 1
fi
```

Comment (deiga) [reply .circleci/test-deploy.yml]: I like that, thanks!

**Comment by deiga** [comment]
@mschile Thanks for fixing it! Could you trigger another run?

**Review by mschile — APPROVED** [review]

**Review by ryanthemanuel — APPROVED** [review]

**Review by ramykl — APPROVED** [review]

**Comment by MikeMcC399** [comment]
@deiga 

The example [src/examples/node-version.yml](https://github.com/cypress-io/circleci-orb/blob/master/src/examples/node-version.yml) from this PR has been merged into the `master` branch and published as release [3.4.0](https://github.com/cypress-io/circleci-orb/releases/tag/v3.4.0), however it has not been published to the website https://circleci.com/developer/orbs/orb/cypress-io/cypress#usage-node-version which still shows the non-working example:

image

- This issue is also described in #468

I don't know how to resolve this issue. It may need help from CircleCI support.

**Pull request merged**